ruby.md

```diff
@@ -5,41 +5,46 @@ Answer the following questions. First, without external resources. Challenge you
 
 1. In what ways are JavaScript and Ruby similar? In what ways are they different?
 
-  Your answer:
+  Your answer:They both use object oriented computer languages, they have built in methods.
 
-  Researched answer:
+  Researched answer:JavaScript is an object-oriented scripting language that allows you to create dynamic HTML pages with interactive effects within a webpage. JavaScript allows you to process and maintain data within the browser. It runs on web-browser only and it is an interpreted programming language. Ruby is an open-source programming language, object-oriented language with a focus on simplicity, portability, extensibility, and productivity. As it is open-source, we can free to modify, edit according to our needs and use.
+  
+  In javascript you use semicolon's in ruby it's not an option. In ruby you use "end" and in java script you use open and closing curly brackets.
+  in Ruby you dont need parenthesis for methods but in java they are required
 
 
 
 2. What is a hash?
 
-  Your answer:
+  Your answer: hashes are like libraries used to store other objects.
 
-  Researched answer:
+  Researched answer: A Hash is created by listing key/value pairs, separated by hash rockets, and enclosed by curly braces.
 
 
 
 3. What is the testing framework used in Ruby? Describe the process of setting up the testing environment.
 
-  Your answer:
+  Your answer: BDD AND TDD testing is used in Ruby. you have to install a gem and then use Rspec.
 
-  Researched answer:
+  Researched answer:By running your Rails tests you can ensure your code adheres to the desired functionality even after some major code refactoring. Rails tests can also simulate browser requests and thus you can test your application's response without having to test it through your browser.
 
 
 
 4. Name three possible relationships between objects?
 
-  Your answer:
+  Your answer: state, class, behavior
 
```

Comment:
There are keywords that define relationships between objects: is-a, has-a, has-many